standards/application/waku-api.md

```diff
@@ -107,90 +107,152 @@ language_mappings:
     module_type: "esm"
 ```
 
-
 ### Type definitions
 
 ```yaml
 types:
   Config:
     type: struct
     fields:
-      operating_mode: 
+      mode:
         type: string
-        constraints: ["edge", "relay"]
-        description: "The mode of operation of the Waku node. Core protocols used by the node are inferred from this mode."
+        constraints: ["client", "server"]
```

Comment:
how is a relay node a "server" node? We had extensive discussions around the topic here.

- client: what do you mean by "client"? the "light client" terminology barely apply to Waku, which is why after several discussions we settled on "edge". because a relay node may also be a "client" for store and peer exchang eservices
- server: how is a node that runs relay, is a "server"?

Note that we are **not** including "Waku as a service node" in this present API.